src/main/java/com/iexec/worker/compute/ComputeManagerService.java

```diff
@@ -34,13 +34,39 @@
 import org.springframework.stereotype.Service;
 
 import java.io.File;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
 
 
 @Slf4j
 @Service
 public class ComputeManagerService {
 
     private static final String STDOUT_FILENAME = "stdout.txt";
+    /**
+     * Examples of timeout depending on category:
+     * <ul>
+     *     <li>XS:   50 * 0.02 = 1 minute</li>
+     *     <li>S :  200 * 0.02 = 4 minutes</li>
+     *     <li>M :  600 * 0.02 = 12 minutes</li>
+     *     <li>L : 1800 * 0.02 = 36 minutes</li>
+     *     <li>XL: 6000 * 0.02 = 120 minutes</li>
+     * </ul>
+     * If computed timeout duration is lower than {@link ComputeManagerService#MIN_PULL_TIMEOUT},
+     * then final timeout duration is {@link ComputeManagerService#MIN_PULL_TIMEOUT}.
+     * <br>
+     * If computed timeout duration is greater than {@link ComputeManagerService#MAX_PULL_TIMEOUT},
+     * then final timeout duration is {@link ComputeManagerService#MAX_PULL_TIMEOUT}.
+     */
+    private static final double PULL_TIMEOUT_FACTOR = 0.02;
+    /**
+     * 1 minute in milliseconds (= 60 000 ms)
+     */
+    private static final long MIN_PULL_TIMEOUT = 1L * 60L * 1000L;
+    /**
+     * 10 minutes in milliseconds (= 600 000 ms)
+     */
+    private static final long MAX_PULL_TIMEOUT = 10L * 60L * 1000L;
```

Comment:
I remember @DamienMure wanted the ability to inject some `IEXEC_WORKER_MAX_PULL_TIMEOUT` from worker env vars.
Hardcoded value for `MIN_PULL_TIMEOUT` seems fine :+1:

Reply:
I think we could set both as external values. This is not lots of work and it could be handy at some point.

Reply:
See https://github.com/iExecBlockchainComputing/iexec-worker/pull/440/commits/167500b241e6cb811086b5ef014cd832aef912b6.